packages/osc-api-auth/src/utils/account.ts

```diff
@@ -328,3 +333,119 @@ export const lmsTokens: LmsTokensFn = async (userId) => {
         },
     });
 };
+
+export const markAsIV: MarkAsIVFn = async (userId, createdBy) => {
+    // Checks for IV role
+    const ivRole = await getRoleByTitle('IV');
+    if (!ivRole) {
+        return new Error('Unable to find IV role');
+    }
+    return await prisma.userRole.create({
+        data: {
+            userId: userId,
+            roleId: ivRole.id,
+            createdBy: createdBy,
+        },
+    });
+};
+
+export const createTutor: CreateTutorFn = async (input, createdBy) => {
+    // Check for existing user, all emails must be unique
+    const existingUser = await getUserByEmail(input.email);
+
+    // If user already exists, throw error
+    if (existingUser) {
+        return new Error('An account already exists for the specified email.');
+    }
+
+    // Create tutor record
+    const user = await prisma.user.create({
+        data: {
+            firstName: input.firstName,
+            lastName: input.lastName,
+            email: input.email,
+        },
+    });
+    if (input.IVUser == true) {
+        await markAsIV(user.id, createdBy);
+    }
+
+    //Assign tutor user role
+    const tutorRole = await getRoleByTitle('Tutor');
+    if (!tutorRole) {
+        return new Error('Unable to verify role');
+    }
+    await prisma.userRole.create({
+        data: {
+            userId: user.id,
+            roleId: tutorRole.id,
+            createdBy: createdBy,
+        },
+    });
+
+    for (var i = 0; i < input.course.length; i++) {
+        await prisma.courseTutor.create({
+            data: {
+                tutorId: user.id,
+                courseId: input.course[i].courseId,
+                createdBy: createdBy,
+                iv: input.course[i].iv,
```

Comment:
Want to consider here that `.iv` may not be submitted if it's a checkbox, so want to type it as optional and then put add a null check

```suggestion
                iv: input.course[i].iv ?? false,
```